### PR TITLE
CLOUDP-110416: add pagination options to agent and server listing

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes

_Jira ticket:_ CLOUDP-110416

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/opsmanager/servers/list.go [hunk 1]
```diff
@@ -49,7 +50,8 @@ var listTemplate = `HOSTNAME	STATE{{range .Results}}
 `
 
 func (opts *ListOpts) Run() error {
-	r, err := opts.store.Agents(opts.ConfigProjectID(), agentType)
+	listOpts := opts.NewListOptions()
+	r, err := opts.store.Agents(opts.ConfigProjectID(), agentType, listOpts)
```
Comment (andreaangiolillo) [inline internal/cli/opsmanager/servers/list.go:54]:
```suggestion
	r, err := opts.store.Agents(opts.ConfigProjectID(), agentType, opts.NewListOptions())
```

File: internal/cli/opsmanager/agents/list.go [hunk 2]
```diff
@@ -48,7 +49,8 @@ var listTemplate = `HOSTNAME	TYPE	STATE{{range .Results}}
 `
 
 func (opts *ListOpts) Run() error {
-	r, err := opts.store.Agents(opts.ConfigProjectID(), opts.agentType)
+	listOpts := opts.NewListOptions()
+	r, err := opts.store.Agents(opts.ConfigProjectID(), opts.agentType, listOpts)
```
Comment (andreaangiolillo) [inline internal/cli/opsmanager/agents/list.go:53]:
```suggestion
	r, err := opts.store.Agents(opts.ConfigProjectID(), opts.agentType, opts.NewListOptions())
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM 💯 . Just a small comment but it is not blocking.

File: internal/cli/opsmanager/servers/list.go (hunk 1 above)
Comment (tibulca) [reply internal/cli/opsmanager/servers/list.go]: done

File: internal/cli/opsmanager/agents/list.go (hunk 2 above)
Comment (tibulca) [reply internal/cli/opsmanager/agents/list.go]: done

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Comment by tibulca** [comment]
documentation was auto-generated

**Pull request merged**